### PR TITLE
feat(braindump): add clear-on-complete preference

### PR DESCRIPTION
**Pull request by ryota-murakami**
## What

Adds a **"Clear finished lines"** preference to the BrainDump section of Preferences. When ON, a finished `- [x] <title>` line is tucked away once its undo window closes, so the scratchpad clears as you go. **Default OFF** — every line stays in place (the on-concept behavior).

## Why

BrainDump is a quick scratchpad. Some people want finished lines to linger (a record of what they did today); others want the page to clear so only open work remains. This makes it a per-user choice instead of a baked-in default.

Per `DESIGN.md` **"Presets First, Then Options"**: keep-every-line ships as the default (self-affirming — the day's work stays visible), and clearing is the opt-in deviation.

## How it works

The clear is tied to the success toast's `onAutoClose`:

- Sonner fires `onAutoClose` **only on the timeout** (the undo window elapsing). An **Undo click** fires `onDismiss` instead — so undone completions are **never** cleared.
- The callback re-resolves the line **by title** against the freshest editor text (the index may have drifted) and removes **only a line that is still checked**. A manual uncheck (or undo) flips the line back to `- [ ]`, which the checked-title lookup won't match — so the clear **self-suppresses**.

No timers, no re-insertion logic, undo flow completely untouched.

## Files

- `schemas/preferences.ts` — `braindumpClearOnComplete: z.boolean().default(false)`
- `redux/slices/preferencesSlice.ts` — reducer + selector + `selectPreferences` broadcast snapshot
- `preferences-sync-channel.ts` — added to the cross-window broadcast allowlist
- `braindump/braindumpUtils.ts` — `removeLineAtIndex` pure helper
- `braindump/BrainDumpEditor.tsx` — `onAutoClose` clear wiring
- `settings/PreferencesSettings.tsx` — the "Clear finished lines" Switch

## Testing

`pnpm validate` green — **602 unit tests pass**, build + typecheck + lint all clean. New behavior coverage:

- **Editor** (3): clears a finished line on auto-close when ON; **keeps** a line that was undone before its window closed (self-suppress); no auto-close hook wired by default. The two non-trivial tests were RED-proven (broke prod code → test failed → restored).
- **Preferences UI** (2): default OFF; opts in when the switch is toggled.
- **Schema / slice / sync** (4): default + explicit opt-in; reducer/selector; cross-window propagation.
- **Util** (3): drops the target line; out-of-range returns original.

## Notes

- No version bump (corelive convention: features land unbumped; release bump is a separate chore).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a “Clear finished lines” preference for the BrainDump editor. When enabled, completed lines are automatically removed after the success notification closes.
* **Bug Fixes**
  * Improved auto-clear behavior to avoid clearing the wrong line when multiple lines are completed or when an undo occurs before the notification closes.
* **Tests**
  * Added/updated unit and integration tests to cover the new preference, default behavior, cross-window syncing, and line-removal edge cases.

### CODE REVIEW

File: src/components/braindump/BrainDumpEditor.tsx [hunk 1]
```diff
@@ -517,9 +522,26 @@ export const BrainDumpEditor = memo(function BrainDumpEditor({
...
+        onAutoClose: clearOnComplete
+          ? () => {
+              const lineToClear = findCheckedLineIndexByTitle(
+                noteTextRef.current,
+                safeTitle,
+              )
+              if (lineToClear === null) return
+              setNoteText(removeLineAtIndex(noteTextRef.current, lineToClear))
+            }
+          : undefined,
```
[the remark this answers is not in the document]
Comment (ryota-murakami) [reply src/components/braindump/BrainDumpEditor.tsx:561]: Fixed in 53f0e5d. Verified against the code: `onAutoClose` left the `checkedRowsRef` entry behind, so with repeated titles + line drift the uncheck path's index/title fallback could delete the wrong `completedId`. The clear is now gated on the completion's `completedId` still being present and deletes that ref entry before mutating text (also self-suppressing across undo / manual uncheck / category-switch). Added a RED-proven regression test.
